determined-ai/determined
# feat: Add metrics-stream/metric-names API for multiple experiments [WEB-1107]

## Description

This PR implements a `/api/v1/experiments/metrics-stream/metric-names` API to support querying metric names for multiple experiments. Additionally, the original `/api/v1/experiments/{experiment_id}/metrics-stream/metric-names` endpoint has been deprecated.





## Test Plan

1. Query the `api/v1/experiments/metrics-stream/metric-names` endpoint with multiple experiment ids, ensure that any metric name is returned at most one time from the API. For example, if all experiments are completed then only the first streaming response will contain metrics. 
2. Run an experiments and query the `api/v1/experiments/metrics-stream/metric-names` endpoint. Ensure that once any metrics are reported they are returned at most on time from the API.
3. Visit the "overview" page for a trial and ensure that metrics are still visible on the chart.
4. Visit the "learning curve" page for an experiment and ensure that metrics are still visible on the chart.



## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

WEB-1107
